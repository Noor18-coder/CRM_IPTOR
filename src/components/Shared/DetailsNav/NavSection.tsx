import React from 'react';
import { useSelector } from 'react-redux';
import { OverlayTrigger } from 'react-bootstrap';
import ImageConfig from '../../../config/ImageConfig';
import * as models from '../../../helpers/Api/models';
import { AppState } from '../../../store/store';

export interface Props {
  backToOpportunityList: () => void;
  popover?: () => void;
}

export const NavSection: React.FC<Props> = ({ backToOpportunityList, popover }) => {
  const popup = popover ? popover() : null;
  const state: models.OpportunityEditOptions = useSelector((appState: AppState) => appState.opportuntyDetails.editOportunity);
  return (
    <>
      <section className="d-flex justify-content-between pagelvl-actionrow">
        <div className="lft-actionitem">
          <img
            src={ImageConfig.LEFT_ARROW}
            onClick={backToOpportunityList}
            onKeyDown={backToOpportunityList}
            role="presentation"
            alt="Back"
            title="Back"
            className="action-icon"
          />
        </div>
        <div className="rgt-actionitem">
          <ul className="list-inline ">
            {
              // Commented Non- Functioning Buttons
              /* <li className="list-inline-item">
                <img src={ImageConfig.HISTORY} alt="History" title="History" />
              </li>
              <li className="list-inline-item">
                <img src={ImageConfig.STAR} alt="Star" title="Star" />
              </li> */
            }
            {popup && state.allowEdit ? (
              <OverlayTrigger rootClose trigger="click" placement="bottom" overlay={popup}>
                <img src={ImageConfig.MORE_V_ELLIPSIS} alt="More" title="More" />
              </OverlayTrigger>
            ) : null}
          </ul>
        </div>
      </section>
    </>
  );
};
