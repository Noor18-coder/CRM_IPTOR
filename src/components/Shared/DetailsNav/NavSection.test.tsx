import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { NavSection } from './NavSection';
import StoreMock from '../../../mocks/Store.mock';

const backToOpportunityList = jest.fn();
const store = StoreMock.createInitialStore();

describe('[Shared] NavSection', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <NavSection backToOpportunityList={backToOpportunityList} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
