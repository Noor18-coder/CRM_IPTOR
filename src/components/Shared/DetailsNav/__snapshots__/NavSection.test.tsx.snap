// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Shared] NavSection should renders correctly 1`] = `
<section
  className="d-flex justify-content-between pagelvl-actionrow"
>
  <div
    className="lft-actionitem"
  >
    <img
      alt="Back"
      className="action-icon"
      onClick={[MockFunction]}
      onKeyDown={[MockFunction]}
      role="presentation"
      src="left-arrow.svg"
      title="Back"
    />
  </div>
  <div
    className="rgt-actionitem"
  >
    <ul
      className="list-inline "
    />
  </div>
</section>
`;
